python/cudf/cudf/core/column/column.py

```diff
@@ -1346,9 +1346,33 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             arbitrary = arbitrary.astype(dtype)
 
         if arb_dtype.kind == "M":
-            data = as_column(pa.array(arbitrary), nan_as_null=nan_as_null)
-            if np.issubdtype(data.dtype, np.datetime64):
+            import datetime as dt
+
+            time_unit, _ = np.datetime_data(arbitrary.dtype)
+            cast_dtype = time_unit in ("D", "W", "M", "Y") or (
+                len(arbitrary) > 0
+                and (
+                    isinstance(arbitrary[0], str)
+                    or isinstance(arbitrary[0], dt.datetime)
+                )
+            )
+
+            if cast_dtype:
+                arbitrary = arbitrary.astype(np.dtype("datetime64[s]"))
+            assert arbitrary.dtype.itemsize == 8
+
+            buffer = Buffer(arbitrary)
+            mask = None
+            if np.any(np.isnat(arbitrary)) or nan_as_null:
```

Comment:
This `np.any(np.isnat(...))` call is really expensive, we should just get rid of it and leave the `nan_as_null` check for now. We can revisit checking for any `NaT` values in the future.

Reply:
Removed this check and currently evaluating based on `nan_as_null`